### PR TITLE
- Simplify vector of vectors

### PR DESCRIPTION
**Pull request by Szymon-Rajczakowski**
- Add const qualifier

## Description

Please include a summary of the change, new sample or fixed issue. Please also include relevant motivation and context.
Please read the [contribution guidelines](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md)

Fixes #<issue number>

## General Checklist:

Please ensure the following points are checked:

- [ ] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [ ] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [ ] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [ ] I have commented any added functions (in line with Doxygen)
- [ ] I have commented any code that could be hard to understand
- [ ] My changes do not add any new compiler warnings
- [ ] My changes do not add any new validation layer errors or warnings
- [ ] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] My changes do not add any regressions
- [ ] I have tested every sample to ensure everything runs correctly
- [ ] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [ ] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [ ] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

### CODE REVIEW

**Comment by Szymon-Rajczakowski** [comment]
Comments:
https://github.com/KhronosGroup/Vulkan-Samples/pull/592#discussion_r1071401204
https://github.com/KhronosGroup/Vulkan-Samples/pull/592/#discussion_r1071404489

**Comment by Szymon-Rajczakowski** [comment]
PR has been updated with additional changes related to these comments:
https://github.com/KhronosGroup/Vulkan-Samples/pull/592/#discussion_r1071394392
https://github.com/KhronosGroup/Vulkan-Samples/pull/592#discussion_r1071395088

File: samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp [hunk 1]
```diff
@@ -764,59 +752,48 @@ glm::vec4 ExtendedDynamicState2::get_changed_alpha(const vkb::sg::PBRMaterial *o
...
 	{
-		const auto &vertex_buffer_pos    = node[i].sub_mesh->vertex_buffers.at("position");
-		const auto &vertex_buffer_normal = node[i].sub_mesh->vertex_buffers.at("normal");
-		auto &      index_buffer         = node[i].sub_mesh->index_buffer;
+		const auto &vertex_buffer_pos    = scene_node[i].sub_mesh->vertex_buffers.at("position");
+		const auto &vertex_buffer_normal = scene_node[i].sub_mesh->vertex_buffers.at("normal");
+		auto       &index_buffer         = scene_node[i].sub_mesh->index_buffer;
 
-		if (scene_index == SCENE_BASELINE_OBJ_INDEX)
+		if (scene_node.at(i).name != "Geosphere")
```
Comment (Karina-Murawko-Wisniewska-Mobica) [inline samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp:788]: Maybe we could use one way of accessing vector scene_node[i] or scene_node.at(i)

**Review by Karina-Murawko-Wisniewska-Mobica — COMMENTED** [review]
LGTM

**Pull request merged**